lib/rubygems/specification_policy.rb

```diff
@@ -371,19 +371,30 @@ def validate_licenses
 
     licenses.each do |license|
       next if Gem::Licenses.match?(license) || license.nil?
+      license_id_deprecated = Gem::Licenses.deprecated_license_id?(license)
+      exception_id_deprecated = Gem::Licenses.deprecated_exception_id?(license)
       suggestions = Gem::Licenses.suggestions(license)
+
+      if license_id_deprecated
+        main_message = "License identifier '#{license}' is deprecated."
+      elsif exception_id_deprecated
+        main_message = "Exception identifier at '#{license}' is deprecated."
+      else
+        main_message = "License identifier '#{license}' is invalid."
+      end
+
       message = <<-WARNING
-license value '#{license}' is invalid.  Use a license identifier from
-http://spdx.org/licenses or '#{Gem::Licenses::NONSTANDARD}' for a nonstandard license,
+#{main_message} Use an identifier from
```

Comment:
So small, but I think the composed message would read slightly better if we keep the final `.` here, instead of as part of `main_message`.

Reply:
I'll fix this and merge.